examples/Nat/ITransportState.cs

```diff
@@ -3,15 +3,30 @@
 
 namespace Pax.Examples.Nat
 {
+  /// <summary>
+  /// Provides a typed interface for the current state of a Transport-layer protocol.
+  /// </summary>
+  /// <typeparam name="T">The type of packet</typeparam>
   internal interface ITransportState<T> where T : Packet
   {
+    /// <summary>
+    /// Gets a boolean value determining if the connection can be removed before the timeout elapses.
+    /// </summary>
     bool CanBeClosed { get; }
 
+    /// <summary>
+    /// Updates the state of the connection to reflect the transmission of the packet.
+    /// </summary>
+    /// <param name="packet">The packet being transmitted</param>
+    /// <param name="packetFromInside">True if the packet originated from inside the NAT, else false.</param>
     void UpdateState(T packet, bool packetFromInside);
   }
 
   internal sealed class NoTransportState<T> : ITransportState<T> where T : Packet
```

Comment:
You're developing a very interesting classification. It would be useful to have a few lines of text at the head of the file, to get a high-level overview of what classes are at play, and how they relate to one another.